tests/test_rtcpeerconnection.py

```diff
@@ -4649,6 +4649,19 @@ async def test_createOffer_without_media(self) -> None:
         # close
         await pc.close()
 
+    @asynctest
+    async def test_setLocalDescription_implicit(self) -> None:
+        pc = RTCPeerConnection()
+        pc.addTrack(AudioStreamTrack())
+        offer = await pc.createOffer()
+        await pc.setRemoteDescription(offer)
+
+        await pc.setLocalDescription()
+        self.assertEqual(pc.localDescription.type, "answer")
+
+        await pc.setLocalDescription()
+        self.assertEqual(pc.localDescription.type, "offer")
+
```

Comment:
This test looks a bit odd, the peer connection is talking.. to itself?

Reply:
Yes. That turns out to be a surprisingly useful test technique!